docs/rfcs/0005-migrate-sitepermission-ac-store-geckoview-store.md

```diff
@@ -0,0 +1,37 @@
+---
+layout: page
+title: Migrate feature-sitepermissions to be compatible with the new GeckoView permission API.
+permalink: /rfc/0005-migrate-sitepermission-ac-store-geckoview-store
+---
+
+* Start date: 2020-07-26
+* RFC PR: [#7855](https://github.com/mozilla-mobile/android-components/pull/7855)
+
+## Summary
+
+Migrate our generic the [SitePermissionsStorage](https://github.com/mozilla-mobile/android-components/blob/56e6f5ba55734014804b3f3b0af5c32414fa1d16/components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsStorage.kt#L33) to be compatible with the new GeckoView permissions API.
+
+## Motivation
+
+For permissions, we currently are storing on disk all the site permissions on AC both for GeckoView and WebView, but that's about to change for GeckoView, a new API is about to land, where GeckoView is going to store all the permissions and we are going to have APIs for managing permissions(query, add, delete, and update).
+
+That put us in a situation where we have to decide on how we are going to approach the migration of the existing user data to the new GeckoView APIs.
+
+So far, I think we have to approaches:
+
+1. Keeping both APIs.
+2. Doing a one-shot migration.
+
+#### 1) Keeping both APIs.
+Continue using the current ac API and do a slow migration where every time we query a site permission, we do it in both stores (GeckoView/AC) and when a new site permission is added, we only add it to the new store the geckoView one, this will cause a long tail transition to the new APIs.
```

Comment:
Yeah, I think that is not really an option since that doesn't really set an end time for the migration.

Reply:
👍

Reply:
Then I will only mention 2) and the approach that we should take :)